### ISSUE TITLE
[Bug]: Table widget breaks when table name is renamed to Number

### ISSUE THREAD

**Issue by shwetha-ramesh** [appsmithorg/appsmith#9189]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

2 issues are observed when renaming the table.
1. Table breaks when it is renamed to a number.
2. When user renames the table to 2 consecutive special characters it successfully renames as '__' 

https://loom.com/share/9930dff46f2141f7bf1167256a8e2566

### Steps To Reproduce

1. Drag and drop table widget
2. In property pane rename the table to number and observe.
2a. In property pane rename the table to 2 consecutive special characters and observe. 

**Expected behavior:** Table must not break when renamed to just number or must show an error message that it is not allowed. 

### Environment

Release

### Version

Cloud

**Comment by areyabhishek** [comment]
@shwetha-ramesh Does this happen for every widget, API, query? Or is this only for the table widget?

**Comment by shwetha-ramesh** [comment]
> @shwetha-ramesh Does this happen for every widget, API, query? Or is this only for the table widget?

I have observed only in table widget.

**Comment by sbalaji1192** [comment]
Numbers are not valid identifiers in javascript. The issue exists on other widgets as well. Widget won't break but we cannot use the widget properties in js expressions.
 we should restrict users from renaming in this case.
somewhat similar to #8881

**Comment by somangshu** [comment]
@sbalaji1192 is right, We should not allow numbers as widget name, I believe that is a platform level change. 

cc @riodeuno

**Comment by somangshu** [comment]
Probably should be a high priority issue and not critical. @ajinkyakulkarni ill leave the call to you

**Comment by riodeuno** [comment]
@somangshu @sbalaji1192 Yes, numbers should not be the first character in identifiers. It is illegal in JS. 
Resolution approach:
- We have functions in place which check for entity name uniqueness. (Explorer/Name.tsx uses this, so does PropertyPaneTitle.tsx)
- We also have sanitiser functions which sanitise the spaces and special characters while entering the entity names (removeSpecialChars is the function name)

Using the above two features, we should disallow naming entities using numbers as the first character.

**Comment by AIshwaryaZemos** [comment]
I did some analysis on this issue and we can add extra check on names if it starts with a number. Please let me know if you are accepting changes in table widgets before I proceed.
